src/dune_rules/dune_package.ml

```diff
@@ -62,6 +62,19 @@ module Lib = struct
       | External ms -> ms
       | Local -> None
     in
+    let melange_runtime_deps =
+      match Lib_info.melange_runtime_deps info with
+      | Local _ -> assert false
+      | External paths ->
+        let lib_dir = Obj_dir.dir obj_dir in
+        List.map
+          ~f:(fun p ->
+            let p =
+              Path.as_in_build_dir_exn p |> Path.Build.drop_build_context_exn
```

Comment:
Can external paths make it here?

Reply:
yeah I guess there's nothing stopping that. What do you think is the best course of action here? Forbid external paths by raising a user_error here?

Reply:
Here would probably be late. You would be able to build the library while its local and only fail when running the install rules. I would think you need to verify this earlier then.

Come to think of it, we shouldn't allow external paths as runtime deps for public libraries at all.

Reply:
> Come to think of it, we shouldn't allow external paths as runtime deps for public libraries at all.

This makes sense to me. Is `install_rules.ml` the best place for that check?

Reply:
I guess so. I would also add it to the corresponding `@check` for the library

Reply:
Added a check for this in 885b58d8ebb772561f852ee88cef4daaba02b58a